### PR TITLE
feat: add `tags_from_metadata` option to LoggerHandler

### LINKED ISSUE #827: Add an ability to set tags from logger metadata
I believe `Sentry.LoggerHandler` would benefit from the option to set Sentry tags based on the log metadata as opposed to using it only as extra context. 

We have a multi-tenant application, where there is a supervision sub-tree per tenant. Each process already correctly tracks their belonging to a tenant using `Logger.metadata` calls, and this tenant ID would be extremely useful to have as a Sentry tag for filtering and grouping. I understand the option of using [`set_tags_context/1`](https://hexdocs.pm/sentry/Sentry.Context.html#set_tags_context/1), but I would rather avoid code duplication given we already set that metadata field, and `LoggerHandler` already reads those metadata keys correctly. 

I think allowing something like `tags: metadata_keys()` option would be enough to cover this use-case.

Thanks a lot!

### PR DESCRIPTION
**Pull request by icehaunter**
Added a new configuration option 'tags_from_metadata' to both `LoggerHandler` that allows specifying which metadata fields should be included as tags in Sentry events. 

Closes #827

### CODE REVIEW

**Comment by ruslandoga** [comment]
Sorry for a drive-by comment, but what happens if metadata values are maps or lists? Wouldn't tags need to be scalar?

File: lib/sentry/logger_backend.ex [hunk 1]
```diff
@@ -141,7 +148,10 @@ defmodule Sentry.LoggerBackend do
 
     sentry_context =
       if sentry_context_from_meta || sentry_context_from_sentry do
-        Map.merge(sentry_context_from_meta || %{}, sentry_context_from_sentry || %{})
+        Map.merge(
+          Map.new(sentry_context_from_meta || %{}),
+          Map.new(sentry_context_from_sentry || %{})
+        )
```
Comment (whatyouhide) [inline lib/sentry/logger_backend.ex:154]: Unnecessary change?

File: lib/sentry/logger_backend.ex [hunk 2]
```diff
@@ -42,6 +42,10 @@ defmodule Sentry.LoggerBackend do
   to `[]`. If set to `:all`, all metadata will be included. `:all` is available
   since v9.0.0 of this library.
 
+  * `:tags` - To include Logger metadata as tags in reports, the `:tags` key can be set to a list of keys.
```
Comment (whatyouhide) [inline lib/sentry/logger_backend.ex:45]: Should we call this `:tags_from_metadata`?

File: lib/sentry/logger_handler.ex [hunk 3]
```diff
@@ -44,6 +44,15 @@ defmodule Sentry.LoggerHandler do
       If set to `:all`, all metadata will be included.
       """
     ],
+    tags: [
+      type: {:or, [{:list, :atom}, {:in, [:all]}]},
+      default: [],
+      type_doc: "list of `t:atom/0`, or `:all`",
+      doc: """
+      Use this to include logger metadata as tags in reports. If it's a list of keys, metadata
+      in those keys will be added as tags to the event. If set to `:all`, all metadata will be included as tags.
```
Comment (whatyouhide) [inline lib/sentry/logger_handler.ex:53]: I don't think we can include **all** metadata as tags because some metadata would be really worthless as tags (`:line` as an example). We should discourage that behavior from users I think. Thoughts?

File: lib/sentry/logger_backend.ex (hunk 2 above)
Comment (whatyouhide) [reply lib/sentry/logger_backend.ex:45]: Also, sorry just seeing now that this is `Sentry.LoggerBackend`: let's not add this option there and make it only available for the Logger handler. We will eventually deprecate and then remove the Logger backend, so better to not add new features to it.

**Review by whatyouhide — CHANGES_REQUESTED** [review]
Some changes but looking great.

File: lib/sentry/logger_handler.ex (hunk 3 above)
Comment (icehaunter) [reply lib/sentry/logger_handler.ex:53]: Yeah, I agree. I'll remove the `:all`

File: lib/sentry/logger_backend.ex (hunk 1 above)
Comment (icehaunter) [reply lib/sentry/logger_backend.ex:154]: Not that it matters given your request to revert changes to the backend, but without this,
`Logger.error("Testing error", sentry: [tags: %{explicit: "tag"}])` code fails even without other changes, as `sentry_context_from_meta` is a list, not a map, so `Map.merge` fails - it's just not covered by existing tests it seems

File: lib/sentry/logger_handler.ex (hunk 3 above)
Comment (icehaunter) [reply lib/sentry/logger_handler.ex:53]: What are your thoughts on `tags` vs `tags_from_metadata` here? I'm OK with either given it lives in `runtime.exs` and isn't repeated often

File: lib/sentry/logger_backend.ex (hunk 1 above)
Comment (whatyouhide) [reply lib/sentry/logger_backend.ex:154]: Aaaah I see sorry, my eyes mixed and I thought this was just a formatting change, I now see that `Map.new/1` is called on both lines. But this is still a fix even if you'll revert the rest of the changes here, won't it? Maybe it would be nice to revert it here and open in as a separate PR with a related regression test case.

File: lib/sentry/logger_handler.ex (hunk 3 above)
Comment (whatyouhide) [reply lib/sentry/logger_handler.ex:53]: `tags_from_metadata` is more explicit I think.

File: lib/sentry/logger_backend.ex (hunk 1 above)
Comment (icehaunter) [reply lib/sentry/logger_backend.ex]: Reverted

**Review by whatyouhide — APPROVED** [review]
Fantastic! 🍬
